test/extended/util/metrics.go

```diff
@@ -0,0 +1,216 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubernetes/pkg/kubectl/util/term"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// pods whose metrics show a larger ratio of requests per
+// second than maxQPSAllowed are considered "unhealthy".
+const maxQPSAllowed = 1.5
+
+type podInfo struct {
+	name      string
+	qps       float64
+	status    string
+	namespace string
+	result    string
+	failed    bool
+}
+
+func calculatePodMetrics(oc *CLI) error {
+	namespaces, err := oc.AdminKubeClient().CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces.Items {
+		// skip namespaces which do not meet "operator namespace" criteria
+		if !strings.HasPrefix(ns.Name, "openshift-") {
+			continue
+		}
+		if !strings.HasSuffix(ns.Name, "-operator") {
+			continue
+		}
+
+		infos, err := getPodInfoForNamespace(oc, ns.Name)
+		if err != nil {
+			return err
+		}
+
+		for _, info := range infos {
+			if info.failed {
+				e2e.Logf("Failed to fetch operator pod metrics for pod %q: %s", info.name, info.result)
```

Comment:
Hmmm... I think we should at least warn about it, or even maybe fail if we can't scrape metrics.

Reply:
Sure, I can aggregate all of these into a single error and return that at the end. Did not want to fail immediately here so as to still be able to collect metrics for any other pods if I fail to scrape on a single one